### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by Irirwanirira**
- [ ] Added the following validations:

## This Pr consists of: 

- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

- [ ] For the Post model:

- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

- [ ] Add unit specs for all of your models' methods and validations.

### Thank you for reviewing my Project

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      subject.posts_counter = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should allow valid values' do
+      subject.posts_counter = 34
+      expect(subject).to_not be_valid
+    end
+  end
+end
```
Comment (ShadyShawkat) [inline spec/models/user_spec.rb]: - [x] Please make sure to test all methods implemented in the `User` model, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods) requirement.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    it 'like_counter should be present' do
+      subject.like_counter = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'comment_count should allow valid values' do
+      subject.comment_count = 12
+      expect(subject).to_not be_valid
+    end
+  end
```
Comment (ShadyShawkat) [inline spec/models/post_spec.rb]: - [x] Please make sure to test all methods implemented in the `Post` model, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods) requirement.

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      @like.author_id = false
+      expect(@like).to_not be_valid
+    end
+
+    it 'when post_id is defined' do
+      @like.post_id = nil
+      expect(@like).to_not be_valid
+    end
+  end
+end
```
Comment (ShadyShawkat) [inline spec/models/like_spec.rb]: - [x] Please make sure to test all methods implemented in the `Like` model, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods) requirement.

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      @comment.author_id = false
+      expect(@comment).to_not be_valid
+    end
+
+    it 'author_id is integer' do
+      @comment.author_id = 'joseph'
+      expect(@comment).to_not be_valid
+    end
+  end
+end
```
Comment (ShadyShawkat) [inline spec/models/comment_spec.rb]: - [x] Please make sure to test all methods implemented in the `Comment` model, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Irirwanirira, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @Irirwanirira , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great request changes application :heavy_check_mark: 
- Great test implementation :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 
- All tests are passed :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**